insights/parsers/nginx_conf.py

```diff
@@ -0,0 +1,335 @@
+    location_statement = space + Optional(modifier) + Optional(space + location + space)
+    if_statement = space + Literal("if") + space + condition + space
+    charset_map_statement = space + Literal("charset_map") + space + value + space + value
+
+    map_statement = space + Literal("map") + space + nonspace + space + dollar_var + space
+    # This is NOT an accurate way to parse nginx map entries; it's almost
+    # certainly too permissive and may be wrong in other ways, but it should
+    # preserve things correctly in mmmmost or all cases.
+    #
+    #    - I can neither prove nor disprove that it is correct wrt all escaped
+    #      semicolon situations
+    # Addresses https://github.com/fatiherikli/nginxparser/issues/19
+    map_pattern = Regex(r'".*"') | Regex(r"'.*'") | nonspace
+    map_entry = space + map_pattern + space + value + space + semicolon
+    map_block = Group(
+        Group(map_statement).leaveWhitespace() +
+        left_bracket +
+        Group(ZeroOrMore(Group(comment | map_entry)) + space).leaveWhitespace() +
+        right_bracket)
+
+    block = Forward()
+
+    # key could for instance be "server" or "http", or "location" (in which case
+    # location_statement needs to have a non-empty location)
+
+    block_begin = (Group(space + key + location_statement) ^
+                   Group(if_statement) ^
+                   Group(charset_map_statement)).leaveWhitespace()
+
+    block_innards = Group(ZeroOrMore(Group(comment | assignment) | block | map_block) + space).leaveWhitespace()
+
+    block << Group(block_begin + left_bracket + block_innards + right_bracket)
+
+    script = OneOrMore(Group(comment | assignment) ^ block ^ map_block) + space + stringEnd
+    return script.parseWithTabs().leaveWhitespace()
+
+
+class UnspacedList(list):
+    """After handling by create_parser(), there is white space existing in the list. Use this class to wrap a list
+     [of lists], making any whitespace entries magically invisible"""
+
+    def __init__(self, list_source):
+        # ensure our argument is not a generator, and duplicate any sublists
+        self.spaced = copy.deepcopy(list(list_source))
+        self.dirty = False
+
+        # Turn self into a version of the source list that has spaces removed
+        # and all sub-lists also UnspacedList()ed
+        list.__init__(self, list_source)
+        for i, entry in reversed(list(enumerate(self))):
+            if isinstance(entry, list):
+                sublist = UnspacedList(entry)
+                list.__setitem__(self, i, sublist)
+                self.spaced[i] = sublist.spaced
+            elif self._spacey(entry):
+                # don't delete comments
+                if "#" not in self[:i]:
+                    list.__delitem__(self, i)
+
+    def _spacey(self, x):
+        return (isinstance(x, str) and x.isspace()) or x == ''
+
+
+@parser('nginx_rh_conf')
+@parser('nginx_org_conf')
+class NginxConf(Parser, LegacyItemAccess):
```

Comment:
Isn't there a problem with applying two specs to the one parser?

If you want to have the `nginx.conf` file found in multiple locations, I think this is where you use the `First` spec, and have only one spec that finds the first file it can in multiple locations.

However, if you want to have multiple files, and a parser for each, then you really need to have one parser class per spec, and one spec per file.  This is where you'd have a main class and then have a simple subclass for each file.